experimental/listener.go

```diff
@@ -37,7 +37,7 @@ type FunctionListener interface {
 	//   - paramValues:  api.ValueType encoded parameters.
 	//
 	// Note: api.Memory is meant for inspection, not modification.
-	Before(ctx context.Context, mod api.Module, def api.FunctionDefinition, paramValues []uint64) context.Context
+	Before(ctx context.Context, mod api.Module, def api.FunctionDefinition, paramValues []uint64, stackIterator api.StackIterator) context.Context
```

Comment:
At that point, `paramValues` is always equivalent to `stackIterator.Next().Args()`. I thought about removing it off the bat, but maybe most people won't care about the whole stack and would prefer an easy access to parameters?

Reply:
I agree with you, let's keep `paramValues` to avoid disrupting existing use cases too much. It would be good to document this tho so nobody is surprised.

Reply:
Would love to get @mathetake opinion on this tho ☝️

Reply:
yeah keeping params sounds good to me